l1-contracts/src/core/Rollup.sol

```diff
@@ -450,20 +472,25 @@ contract Rollup is EIP712("Aztec Rollup", "1"), Leonidas, IRollup, ITestRollup {
    * @param _archive - A root of the archive tree after the L2 block is applied
    * @param _blockHash - The poseidon2 hash of the header added to the archive tree in the rollup circuit
    * @param _signatures - Signatures from the validators
-   * @param _body - The body of the L2 block
+   * // TODO(#9101): The below _body should be removed once we can extract blobs. It's only here so the archiver can extract tx effects.
+   * @param  - The body of the L2 block
+   * @param blobInput - The blob evaluation KZG proof, challenge, and opening required for the precompile.
    */
   function propose(
     bytes calldata _header,
     bytes32 _archive,
     bytes32 _blockHash,
     bytes32[] memory _txHashes,
     SignatureLib.Signature[] memory _signatures,
-    bytes calldata _body
+    // TODO(#9101): Extract blobs from beacon chain => remove below body input
+    bytes calldata,
+    bytes calldata blobInput
   ) public override(IRollup) {
     if (canPrune()) {
       _prune();
     }
-    bytes32 txsEffectsHash = TxsDecoder.decode(_body);
+
+    bytes32 blobHash = _validateBlob(blobInput);
```

Comment:
See later comment for more context. But should add a comment along the lines.
```solidity
// Since an invalid blob hash here would fail the consensus checks of 
// the header, the `blobInput` is implicitly accepted by consensus as well.
```

Reply:
Not sure I understand this one - yes, the first 32 bytes (blobHash) would be accepted, but the remainder of the `blobInput` needs to be correct (valid proof, opening, etc) for it to be accepted here.

Reply:
It is ok if it "breaks" here. The thing you don't want is that it can pass here but fail at the proof while having an honest committee. So since the same values are used here and in the proof, if they are good here they should also be good later.